src/lru/test.rs

```diff
@@ -0,0 +1,89 @@
+#![cfg(test)]
+
+use super::*;
+use linked_hash_map::LinkedHashMap;
+
+#[derive(Debug)]
+struct TestNode {
+    id: usize,
+    index: LruIndex,
+}
+
+impl TestNode {
+    fn new(id: usize) -> Arc<Self> {
+        Arc::new(TestNode {
+            id,
+            index: Default::default(),
+        })
+    }
+}
+
+impl LruNode for TestNode {
+    fn lru_index(&self) -> &LruIndex {
+        &self.index
+    }
+}
+
+const LRU_SEED: &str = "Hello, Rustaceans";
+const PICK_SEED: &str = "Wippity WIP";
+
+/// Randomly requests nodes and compares the performance of a
+/// *perfect* LRU vs our more approximate version. Since all the
+/// random number generators use fixed seeds, these results are
+/// reproducible. Returns (oracle_hits, lru_hits) -- i.e., the number
+/// of times that the oracle had something in cache vs the number of
+/// times that our LRU did.
+fn compare(num_nodes: usize, capacity: usize, requests: usize) -> (usize, usize) {
```

Comment:
Hm, frankly speaking, this does not make sense to me :-) Either I miss something crucial,  or this indeed is not a useful comparison. 

I think if you pick requests **at random**, than eviction strategy is completely irrelevant, because the probability of each hit is independent of previous requests and equals to `lru_size / total_size`. In other words, if requests are uniformly and independently distributed, the contents of lru is completely irrelevant for hit/miss probability, only the size matters.

It only makes sense to measure LRU performance with respect to a particular request distribution which exhibits temporal locality.

Reply:
Yeah, you're not wrong. I wanted to introduce a normal distribution or something like that but the rand crate for doing so looked a bit complicated and I never got around to writing the code. The RLS tests however seem to suggest it works pretty decently. But it'd be good indeed to connect this to a proper distribution and see what it looks like. (In particular, I think there is some more tuning we could do to achieve a better hit-rate.)

Reply:
Create tests using normal distributions and various standard deviations. Approx. LRU performs very well.